### PR TITLE
feat: implement Vertex AI (Gemini) api-translation provider

### PR DESCRIPTION
**Pull request by asaadbalum**
Add Vertex AI provider to the api-translation plugin, translating OpenAI Chat Completions requests to Google Gemini GenerateContent API format and responses back to OpenAI schema.

## Description

Implements the Vertex AI (Gemini) api-translation provider (#11), following the same Provider interface pattern used by the Anthropic and Azure OpenAI translators.

**Request translation (TranslateRequest):**
- Converts OpenAI messages array to Gemini `contents` format
- Maps `system`/`developer` role messages to `systemInstruction`
- Maps `assistant` role to `model` role
- Translates generation parameters (`max_tokens`/`max_completion_tokens` → `maxOutputTokens`, `temperature`, `top_p` → `topP`, `stop` → `stopSequences`)
- Rewrites `:path` to `/v1beta/models/{model}:generateContent`
- Validates model name to prevent path injection
- Sets `content-type: application/json` header

**Response translation (TranslateResponse):**
- Converts Gemini `candidates` to OpenAI `choices`
- Maps Vertex `finishReason` to OpenAI `finish_reason` (STOP→stop, MAX_TOKENS→length, SAFETY/RECITATION/BLOCKLIST/PROHIBITED_CONTENT/SPII/MODEL_ARMOR/IMAGE_SAFETY/IMAGE_PROHIBITED_CONTENT/IMAGE_RECITATION→content_filter, MALFORMED_FUNCTION_CALL/UNEXPECTED_TOOL_CALL→tool_calls)
- Translates `usageMetadata` to OpenAI `usage` format
- Converts `functionCall` parts to OpenAI `tool_calls` with generated IDs
- Translates Vertex error responses to OpenAI error format
- Gracefully handles missing `usageMetadata`, empty `candidates`, missing `content.parts`

**Registration and integration:**
- Adds `Vertex = "vertex"` constant to `pkg/external-model/provider/provider.go`
- Registers the Vertex provider in the api-translation plugin alongside Anthropic and Azure OpenAI
- Adds Vertex to the apikey-injection plugin's `defaultInjectors` with `Authorization: Bearer` format (same as OpenAI)

**API target — v1beta:**
The translator targets `/v1beta/models/{model}:generateContent` because `systemInstruction` requires `v1beta` on the Generative Language API (`generativelanguage.googleapis.com`). The Vertex AI API (`aiplatform.googleapis.com`) supports `systemInstruction` in `v1`, but `v1beta` is compatible with both endpoints. If the project moves to the enterprise Vertex AI API, the path format would change entirely — that path transformation would likely be handled by Envoy path rewriting, not by this plugin.

## How Has This Been Tested?

- **Unit tests** (`vertex_test.go` — 60+ tests): basic translation, system messages, multi-turn conversations, all generation parameters, all finish reason mappings, function calling, error responses, model name validation, edge cases (missing fields, empty content, malformed structures, unknown fields), helper function coverage
- **Unit test coverage**: 100% statement coverage on the Vertex provider
- **Plugin integration tests** (`plugin_test.go` — 4 new tests): `TestProcessRequest_VertexProvider`, `TestProcessResponse_Vertex`, `TestProcessResponse_VertexError`, `TestProcessResponse_VertexToolCall`
- **apikey-injection tests**: Vertex injector test added (`TestProcessRequest` Vertex case, `TestDefaultInjectors` updated)
- **Full test suite**: all existing tests (Anthropic, Azure OpenAI, model-provider-resolver, apikey-injection) continue to pass
- **Real Vertex AI API validation**: tested against Google's Generative Language API with multiple models (gemini-2.5-flash, gemini-2.0-flash-lite, gemini-2.5-pro), including system instruction, multi-turn, function calling, MAX_TOKENS truncation, and error scenarios
- **Simulator validation**: tested against the team's llm-katan mock server — full pipeline (OpenAI request → TranslateRequest → simulator → TranslateResponse → OpenAI response) verified
- **E2E validation**: BBR server deployed in kind cluster with MaaSModelRef CRD, model-provider-resolver, api-translation, and apikey-injection plugins — full gRPC ext_proc pipeline confirmed working with Vertex translation (request body mutation, header mutations, response body mutation, API key injection)
- **Lint**: `golangci-lint run ./...` — 0 issues

**Test evidence (real API round-trip):**
```
# OpenAI request → our TranslateRequest → translated Gemini body
{"contents":[{"parts":[{"text":"What is Kubernetes?"}],"role":"user"}],
 "generationConfig":{"maxOutputTokens":100,"temperature":0.1},
 "systemInstruction":{"parts":[{"text":"Be concise. One sentence max."}]}}

# Real Gemini API response:
{"candidates":[{"content":{"parts":[{"text":"Kubernetes is an open-source
 platform for automating the"}],"role":"model"},"finishReason":"MAX_TOKENS"}],
 "usageMetadata":{"promptTokenCount":12,"candidatesTokenCount":11,
 "totalTokenCount":108}}

# Our TranslateResponse → OpenAI format:
{"choices":[{"finish_reason":"length","index":0,"message":{"content":
 "Kubernetes is an open-source platform for automating the",
 "role":"assistant"}}],"model":"gemini-2.5-flash",
 "object":"chat.completion","usage":{"completion_tokens":11,
 "prompt_tokens":12,"total_tokens":108}}
```

**E2E evidence (kind cluster, full gRPC pipeline):**
```
# Request body mutation — OpenAI → Gemini format:
Body mutation: {"contents":[{"parts":[{"text":"What is Go?"}],"role":"user"}],
 "generationConfig":{"maxOutputTokens":50,"temperature":0.5},
 "systemInstruction":{"parts":[{"text":"Be concise."}]}}
Header mutations: SET :path /v1beta/models/gemini-2.5-flash:generateContent
                  SET Authorization: Bearer <key>
                  REMOVE authorization
Server log: "API key injected","provider":"vertex"

# Response body mutation — Gemini → OpenAI format:
{"choices":[{"finish_reason":"stop","index":0,"message":{"content":
 "Go is a statically typed compiled language.","role":"assistant"}}],
 "model":"gemini-2.5-flash","object":"chat.completion",
 "usage":{"completion_tokens":8,"prompt_tokens":10,"total_tokens":18}}
```

## Summary by CodeRabbit

* **New Features**
  * Added Google Vertex AI as a supported provider ("vertex") with automatic translation between OpenAI and Vertex request/response formats (system instructions, tool calls, generation params, error mapping) and built-in API key injection using Authorization: Bearer.

* **Tests**
  * Added comprehensive tests covering Vertex request/response translation, validation, error scenarios, parameter mapping, and API key injection behavior.

### CODE REVIEW

File: pkg/plugins/api-translation/providers/vertex/vertex.go [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+)
+
+const (
+	vertexV1BetaPathTemplate = "/v1beta/models/%s:generateContent"
```
Comment (nirrozenbaum) [inline pkg/plugins/api-translation/providers/vertex/vertex.go]: why `v1beta`?
I see here `v1`:
https://docs.cloud.google.com/vertex-ai/generative-ai/docs/reference/rest/v1/projects.locations.publishers.models/generateContent
